### PR TITLE
Allow overriding HwId data from a config file

### PR DESCRIPTION
**Pull request by hughsie**
To use this create a file in /var/lib/fwupd/hwids.conf

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — COMMENTED** [review]
I guess this is an easier to use version of `FWUPD_SYSFSFWDIR`.  I think it makes sense, but shouldn't it be in `/etc`?

File: libfwupdplugin/tests/hwids.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[fwupd]
```
Comment (superm1) [inline libfwupdplugin/tests/hwids.conf:1]: does this need to be distributed in `fwupd-test` binary packages?

**Comment by hughsie** [comment]
>  I think it makes sense, but shouldn't it be in /etc

I did ponder adding this to `daemon.conf` and then also supporting loading from `/usr/lib/fwupd/daemon.conf` as an override.

**Comment by superm1** [comment]
> >  I think it makes sense, but shouldn't it be in /etc
> 
> I did ponder adding this to `daemon.conf` and then also supporting loading from `/usr/lib/fwupd/daemon.conf` as an override.

That feels more correct to me.

**Comment by hughsie** [comment]
> That feels more correct to me.

Agreed; PR updated.

File: data/daemon.conf [hunk 2]
```diff
@@ -48,3 +48,7 @@ IgnorePower=false
 
 # Only support installing firmware signed with a trusted key
 OnlyTrusted=true
+
+# these are only required when the SMBIOS or Device Tree data is invalid or missing;
+# the full list can be found here: https://fwupd.github.io/libfwupdplugin/
```
Comment (superm1) [inline data/daemon.conf:53]: why not just list them all?  This is (relatively) static and there are only 13 of them.

Comment (hughsie) [reply data/daemon.conf:53]: I think it's actually better than that; we only want to override the non-version ones. Patch updated.

**Review by superm1 — COMMENTED** [review]
Just one idea, otherwise LGTM now

File: libfwupdplugin/fu-hwids.c [hunk 3]
```diff
@@ -351,6 +351,50 @@ fu_hwids_convert_integer_cb(FuSmbios *smbios, guint8 type, guint8 offset, GError
...
+	/* optional overrides */
+	localstatedir = fu_common_get_path(FU_PATH_KIND_LOCALSTATEDIR_PKG);
+	fn_var = g_build_filename(localstatedir, "daemon.conf", NULL);
+	if (g_file_test(fn_var, G_FILE_TEST_EXISTS)) {
+		g_debug("loading HwId overrides from %s", fn_var);
+		if (!g_key_file_load_from_file(kf, fn_var, G_KEY_FILE_NONE, error))
+			return FALSE;
+	} else {
+		g_debug("not loading HwId overrides from %s", fn_var);
+	}
```
[the remark this answers is not in the document]
Comment (hughsie) [reply libfwupdplugin/fu-hwids.c]: Good idea, something like that^^?

**Review by superm1 — APPROVED** [review]
Yup, looks great!

**Pull request merged**